src/framework/mpas_io.F

```diff
@@ -5733,6 +5880,14 @@ subroutine MPAS_io_put_att_real0d(handle, attName, attValue, fieldname, syncVal,
          if (present(ierr)) ierr = MPAS_IO_ERR_BACKEND
          return
       end if
+
+!      if (handle % preexisting_file) then
+!         pio_ierr = PIO_enddef(handle % pio_file)
+!         if (pio_ierr /= PIO_noerr) then
+!            io_global_err = pio_ierr
+!            return
+!         end if
+!      end if
```

Comment:
Perhaps it's better to just remove this block of code rather than leaving it commented-out?